File: src/main/java/Library/Authentication.java
package Library;

//  Service: handling the login/out

public class Authentication {
    private final Accounts accounts;
    private Borrower currentUser;

    public Authentication(Accounts accounts){
        this.accounts = accounts;
    }

    public boolean login(String username, String password) {
        for (Borrower b : accounts.all()) {
            if (b.getUsername().equals(username) && b.getPassword().equals(password)) {
                currentUser = b;
                return true;
            }
        }
        return false;
    }

    public Borrower getCurrentUser() {
        return currentUser;
    }
}

File: A3/models/Authentication.js
// Authentication.js

class Authentication {
    constructor(accounts) {
        this.accounts = accounts;
        this.currentUser = null;
    }

    // login(String username, String password)
    login(username, password) {
        const allUsers = this.accounts.all();

        for (const borrower of allUsers) {
            if (
                borrower.getUsername() === username &&
                borrower.getPassword() === password
            ) {
                this.currentUser = borrower;
                return true;
            }
        }
        return false;
    }

    // getCurrentUser()
    getCurrentUser() {
        return this.currentUser;
    }

    // logout()
    logout() {
        this.currentUser = null;
    }
}

module.exports = Authentication;
